PocketKit/Sources/PocketKit/MyList/SavesContainerViewController.swift

```diff
@@ -369,6 +378,14 @@ extension SavesContainerViewController {
             self?.present(viewModel: addTagsViewModel)
         }.store(in: &readableSubscriptions)
 
+        readable.$presentedListenViewModel
+            .filter { $0 != nil }
+            .sink { [weak self] listenViewModel in
+                guard let listenViewModel else { return }
+                self?.showListen(listenViewModel: listenViewModel)
+                readable.presentedListenViewModel = nil
+            }.store(in: &subscriptions)
```

Comment:
This is a quirky thing to do. Basically, this is a publisher-as-a-delegate. This is done so that we can _use_ the requested view model to kickstart Listen, but the "host" view model (in this highlighted case, the reader view) doesn't really need to maintain a reference to it, only supply it. To not retain the view model here, we immediately nil-out the supplied view model.

What's a more appropriate way?

Reply:
Cleaned up in a followup commit.